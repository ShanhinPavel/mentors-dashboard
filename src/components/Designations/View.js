import React from 'react';
import PropTypes from 'prop-types';
import {
  designationContainer, desinationTitle, listContainer, listTitle,
} from '../../styles/designations';

function View(props) {
  const { tasksMarkers, studentMarkers } = props;

  return (
    <div style={designationContainer}>
      <h2 style={desinationTitle}>Designations</h2>
      <div style={listContainer}>
        <div>
          <h3 style={listTitle}>Task marking</h3>
          {tasksMarkers}
        </div>
        <div>
          <h3 style={listTitle}>Task marking</h3>
          {studentMarkers}
        </div>
      </div>
    </div>
  );
}

View.propTypes = {
  tasksMarkers: PropTypes.objectOf(PropTypes.node).isRequired,
  studentMarkers: PropTypes.objectOf(PropTypes.node).isRequired,
};


export default View;
