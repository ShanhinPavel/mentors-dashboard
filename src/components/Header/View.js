import React from 'react';
import PropTypes from 'prop-types';
import { containerStyle } from '../../styles/header';
import Login from '../Log-in';
import Logout from '../Log-out';

function View(props) {
  const {
    data, onClickLogin, onClickLogout, handleChange,
  } = props;
  // const { authStatus, userName } = data;
  const { userName, userStatus } = data;

  return (
    <div className="header" style={containerStyle}>
      <h2 style={{ marginLeft: 10 }}>Mentor dashboard</h2>
      {userStatus === 'unauthorised'
        ? <Login onClickLogin={onClickLogin} change={handleChange} />
        : <Logout onClickLogout={onClickLogout} userName={userName} />
          }
    </div>
  );
}

View.propTypes = {
  onClickLogout: PropTypes.func.isRequired,
  onClickLogin: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
};


export default View;
