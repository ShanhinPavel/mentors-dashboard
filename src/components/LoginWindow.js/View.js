import React from 'react';

const loginContainer = {
  width: 200,
  height: 200,
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
  zIndex: 200,
};

const darkSide = {
  background: 'rgba(255, 255, 255, 0.5)',
  backgroundSize: 'cover',
  zIndex: 100,
};

function View(props) {
  return (
    <div style={darkSide}>
      <div style={loginContainer}>
        <button type="submit">Login</button>
      </div>
    </div>
  );
}

export default View;
