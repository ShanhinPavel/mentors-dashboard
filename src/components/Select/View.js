import React from 'react';
import { arrayOf, node, func } from 'prop-types';
import { selectContainer, mentorP, select } from '../../styles/select';

function View(props) {
  const { onChange, optionElements } = props;

  return (
    <div style={selectContainer}>
      <p style={mentorP}>Mentor</p>
      <select id="great-names" onChange={onChange} style={select}>
        <option>
            Choose mentor
        </option>
        {optionElements}
      </select>
    </div>
  );
}

View.propTypes = {
  optionElements: arrayOf(node).isRequired,
  onChange: func.isRequired,
};

export default View;
