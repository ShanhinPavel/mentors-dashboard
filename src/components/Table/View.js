/* eslint-disable max-len */
import React from 'react';
import {
  objectOf, any, arrayOf, object,
} from 'prop-types';
import {
  table, tableHeader, headerTaskColumn, headerStudentsColumn, tableBody, tasksColumn, studentsColumn,
} from '../../styles/table';

function View(props) {
  const { rowsOfTasks, rowsOfStudentTasks } = props;

  return (
    <div className="table" style={table}>

      <div className="table__header" style={tableHeader}>
        <div className="header__task-column" style={headerTaskColumn}>
          <h3>Tasks</h3>
        </div>
        <div className="header__students-column" style={headerStudentsColumn}>
          <h3>Students</h3>
        </div>
      </div>

      <div className="table__body" style={tableBody}>
        <div className="tasksColumn" style={tasksColumn}>
          {rowsOfTasks}
        </div>
        <div className="students-rows" style={studentsColumn}>
          {rowsOfStudentTasks}
        </div>
      </div>
    </div>
  );
}

View.propTypes = {
  rowsOfTasks: objectOf(any).isRequired,
  rowsOfStudentTasks: arrayOf(object).isRequired,
};

export default View;
