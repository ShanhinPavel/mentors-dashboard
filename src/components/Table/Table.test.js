/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import View from './View';

describe('Table', () => {
  test('Should render "View" correctly in "debug" mode', () => {
    const component = shallow(<View debug />);
    expect(component).toMatchSnapshot();
  });
});
