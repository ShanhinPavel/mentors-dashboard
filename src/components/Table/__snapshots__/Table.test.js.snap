// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Should render "View" correctly in "debug" mode 1`] = `
<div
  className="table"
  style={
    Object {
      "backgroundColor": "white",
      "border": "2px solid black",
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": "20px",
      "minWidth": 320,
      "width": "fit-content",
    }
  }
>
  <div
    className="table__header"
    style={
      Object {
        "borderBottom": "2px solid black",
        "display": "flex",
        "flexDirection": "row",
        "minHeight": 30,
      }
    }
  >
    <div
      className="header__task-column"
      style={
        Object {
          "borderRight": "2px solid black",
          "minWidth": 220,
          "textAlign": "center",
          "widht": "44%",
        }
      }
    >
      <h3>
        Tasks
      </h3>
    </div>
    <div
      className="header__students-column"
      style={
        Object {
          "minWidth": 100,
          "textAlign": "center",
          "width": "66%",
        }
      }
    >
      <h3>
        Students
      </h3>
    </div>
  </div>
  <div
    className="table__body"
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <div
      className="tasksColumn"
      style={
        Object {
          "borderRight": "2px solid black",
          "display": "flex",
          "flexDirection": "column",
          "minWidth": 220,
        }
      }
    />
    <div
      className="students-rows"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "minWidth": 100,
        }
      }
    />
  </div>
</div>
`;
